### PR TITLE
Text-wrap by default for command line arguments

### LINKED ISSUE #8034: Avoid horizontal scroll bar in launch profile arguments text box
#7919 allowed newlines in launch profile arguments for executables.

If you have one very long argument then a horizontal scroll bar is introduced, making the text box harder to work with.

Anecdotally, launch arguments are often file paths, which tend to be lengthy and can end up looking this like:

image

We should investigate wrapping text when it overflows here.

When addressing this, we should consider whether we want to control wrapping on a per-property basis. For example, we may not wish to have wrapping for build pre/post events, which tend to contain executable code. Executable code does not format well with wrapping.

### PR DESCRIPTION
**Pull request by adamint**
Fixes #8034 (for good this time).

See https://devdiv.visualstudio.com/DevDiv/_git/CPS/pullrequest/455702 for more details

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8893)

### CODE REVIEW

**Review by MiYanni — APPROVED** [review]

File: docs/repo/property-pages/property-specification.md [hunk 1]
```diff
@@ -190,6 +190,8 @@ If the text within this editor should be displayed with a monospace (fixed width
 </StringProperty>
 ```
 
+Multi-line strings can optionally set metadata to control text wrapping, where the value corresponds to an enum value of [TextWrapping](https://learn.microsoft.com/en-us/dotnet/api/system.windows.textwrapping?view=windowsdesktop-7.0). Example: `<NameValuePair Name="TextWrapping" Value="Wrap" />`. Text-wrapping defaults to NoWrap.
```
Comment (MiYanni) [inline docs/repo/property-pages/property-specification.md:193]: Don't have `en-us` in the url.
`https://learn.microsoft.com/dotnet/api/system.windows.textwrapping?view=windowsdesktop-7.0`

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ProjectDebugPropertyPage.xaml [hunk 2]
```diff
@@ -31,6 +31,7 @@
       <ValueEditor EditorType="MultiLineString">
         <ValueEditor.Metadata>
           <NameValuePair Name="UseMonospaceFont" Value="True" />
+          <NameValuePair Name="TextWrapping" Value="Wrap" />
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ProjectDebugPropertyPage.xaml:34]: I think we also need this in the executable launch profile XAML.

**Review by drewnoakes — APPROVED** [review]

File: docs/repo/property-pages/property-specification.md (hunk 1 above)
Comment (drewnoakes) [reply docs/repo/property-pages/property-specification.md:193]: Also use the `netframework-4.7.2` version as that's the version of WPF we target in VS.

```suggestion
Multi-line strings can optionally set metadata to control text wrapping, where the value corresponds to an enum value of [TextWrapping](https://learn.microsoft.com/dotnet/api/system.windows.textwrapping?view=netframework-4.7.2). Example: `<NameValuePair Name="TextWrapping" Value="Wrap" />`. Text-wrapping defaults to `NoWrap`.
```

**Review by melytc — APPROVED** [review]
Thanks for adding documentation, too 🙂

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ProjectDebugPropertyPage.xaml (hunk 2 above)
Comment (adamint) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ProjectDebugPropertyPage.xaml:34]: Yup, you're right.

**Comment by melytc** [comment]
Saw this yesterday while debugging in VS:
text-wrap

Would be nice in the future to have this option 🙂

**Review by melytc — APPROVED** [review]

**Pull request merged**